PostProcessing/master.py

```diff
@@ -91,22 +102,71 @@ def HL_fitting_function(infile, outfilename, func_name="MultiGauss",
         workers.close()
 
         # Unravel results into output grids
-        params, obs_err, chi_grid = Posproc.results_to_grid(results, len(lats), len(lons))
+        params, obs_err, rss_func_grid, rss_mean_grid, dof = (
+                        Posproc.results_to_grid(results, len(lats), len(lons)))
 
         # Plot some results if requested
         if plot != None:
            print("MESSAGE: Plotting results - data versus fitting: " + str(depth[lev]) + " m")
            Plots.plot_data_vs_fitting(outfig, plot, x_val, cors, var, obs_err, lats, lons,
                                        depth[lev], params, func_name, num_funcs, lenscale)
-                                      
+
+        # If requested perform F-test comparing to mean
+        if f_test:
+            if (func_name == "MultiGauss"):
+                num_params = 2 * num_funcs
+            elif (func_name == "MultiGauss_Fixed"):
+                num_params = num_funcs
+            else:
+                raise ValueError("Cannot calculate num_params for "
+                                 + f"function={func_name}")
+            p_val = f_test_pvalue(rss_func_grid, rss_mean_grid, num_params,
+                           num_params + dof)
+        else:
+            p_val = None
```

Comment:
Minor comment here - perhaps you can get rid of the else statement and move p_val=none to line 114.

Reply:
Done.